scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -582,7 +608,12 @@ def check_containers( self, run_state ):
                 k8s_data = self.get_k8s_data()
 
                 self._logger.log(scalyr_logging.DEBUG_LEVEL_2, 'Attempting to retrieve list of containers:' )
-                running_containers = _get_containers(self.__client, ignore_container=self.container_id, glob_list=self.__glob_list, include_log_path=True, include_k8s_info=True)
+
+                running_or_finished_since = None
+                if self.__short_lived_seconds is not None:
```

Comment:
At one point, we had discussed checking for all containers that were created after the last time we polled the container list (and are already dead).  I forget, why can't we do that?

Reply:
Hmm.. in my mind, this points at a larger problem.  What happens if a container we are watching exits?  Do we make sure we finish copying its logs, or do we stop where ever we were when the container exits?  It would be better for us to put such a container in a state of "keep copying until you reach the end of the log file and then no longer watch the log file."